update-version.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+set -e
+
+# This script is updating AAT Library version, it is looking for regex pattern `(ablyAssetTrackingLibraryVersion = ").+(")` and replace 
+# the value of `ablyAssetTrackingLibraryVersion` with the new value ($1) passed to the script.
+# This script starts searching in the project root directory recursively and processes all files in the project in search of the pattern above.
+#
+# usage: `./update-version.sh 1.0.0-beta.5`
+
+find ./ -type f -and -name "*.swift" -print0 | xargs -0 sed -i.old_version -E 's/(ablyAssetTrackingLibraryVersion = ").+(")/\1'$1'\2/'
```

Comment:
I'm not a fan of manual changing the version before release. As long as it makes sense when you have to change it in one place in so far manual changing this value in more than one place could cause errors.
I named the variable `ablyAssetTrackingLibraryVersion` for a purpose, the script is looking for this name in the whole codebase and there is a higher probability there is more than one variable called `libraryVersion` which is not a version of AAT.

Reply:
> I named the variable `ablyAssetTrackingLibraryVersion` for a purpose, the script is looking for this name in the whole codebase and there is a higher probability there is more than one variable called `libraryVersion` which is not a version of AAT.

I understand the motivation for naming the variable `ablyAssetTrackingLibraryVersion`. The point I was trying to make is that the variable _isn't_ named `ablyAssetTrackingLibraryVersion` [in the Swift code](https://github.com/ably/ably-asset-tracking-swift/blob/f68de7515d7e61cc9a1f358d6e0b5c489523c0f9/Sources/AblyAssetTrackingCore/Version.swift) – it's called `libraryVersion`.

> I'm not a fan of manual changing the version before release. As long as it makes sense when you have to change it in one place in so far manual changing this value in more than one place could cause errors.

Sorry, I don't think I understand what you're saying here. What do you mean by "manual changing the version before release"? My suggestion was just around modifying this script to just target the specific files that contain the version number, principally to avoid the exact problem you mentioned above, which is accidentally touching unrelated variables also named `libraryVersion`.

Reply:
> Sorry, I don't think I understand what you're saying here. What do you mean by "manual changing the version before release"? My suggestion was just around modifying this script to just target the specific files that contain the version number, principally to avoid the exact problem you mentioned above, which is accidentally touching unrelated variables also named `libraryVersion`.

oh, my bad, now I get it. I'll change it to point out specific files ane I'll rename the variable

Reply:
@lawrence-forooghian here ab42ce82cf55932f0da3450e2460e4ca3acd8fe9

Reply:
I still don't understand how it works, does it look for `s/(ibraryVersion = ")` _without leading 'l'_ ? `Version.swift` still doesn't contain `ablyAssetTrackingLibraryVersion` var.

Reply:
> I still don't understand how it works, does it look for `s/(ibraryVersion = ")` _without leading 'l'_ ? `Version.swift` still doesn't contain `ablyAssetTrackingLibraryVersion` var.

yea, I didn't notice there is no `l` in var name. Of course, it should be named `libraryVersion`, it's not called `ablyAssetTrackingLibraryVersion` anymore:
77a683fcce6ba201a43bff6aa41e762f93ca6438

Reply:
@lawrence-forooghian could you resolve this conversation if it's ok?